src/jit/compiler.h

```diff
@@ -661,23 +701,23 @@ class  LclVarDsc
     bool                IsFloatRegType() const
                         {
                             return
-#ifdef _TARGET_ARM_
+#ifdef FEATURE_HFA
                                 lvIsHfaRegArg ||
 #endif
                                 isFloatRegType(lvType);
                         }
-#ifdef _TARGET_ARM_
+#ifdef FEATURE_HFA
     var_types           GetHfaType() const
                         {
-                            assert(lvIsHfaRegArg);
-                            return lvIsHfaRegArg ? (lvHfaTypeIsFloat ? TYP_FLOAT : TYP_DOUBLE) : TYP_UNDEF;
+                            assert(lvIsHfa());
+                            return lvIsHfa() ? (lvHfaTypeIsFloat ? TYP_FLOAT : TYP_DOUBLE) : TYP_UNDEF;
```

Comment:
Callers are all either guarded to expect to recieve TYP_UNDEF:

 hfaType    = compiler->GetHfaType(objClass);   // set to float or double if it is an HFA, otherwise TYP_UNDEF
   return varTypeIsFloating(GetHfaType(hClass));